### ISSUE TITLE
[Android 9 - TargetSKD 28] Backbutton not working

### ISSUE THREAD

**Issue by fgarcia5** [apache/cordova-android#687]
Hi all.

I am developing an App with Cordova.

The behavior of the back button on my Android 9 device is not what I expected:
Setting the targetSDK to 28 (from Android Studio >> Project Structure >> app >> Flavors), the backbutton listener doesn't work, it kills the Application directly.
On the other hand, if I make a downgrade of the targetSDK, the backbutton works correctly.

This behavior only occurs on devices with Android 9 and project targetSDK level 28. In other devices with other versions of Android it works correctly.

This way I'm subscribing to the backbutton event:
```
document.addEventListener("deviceready", onDeviceReady, false);
function onDeviceReady() 
{   
	document.addEventListener("backbutton", onBackKeyDown, false);
}
function onBackKeyDown() 
{
	//Do something...
}
```
I have tried with different versions of cordova, cordova-android and Android Studio but I have not been able to solve the problem.

Somebody could help me?

Thank you!

**Comment by hazems** [comment]
I tried this personally and I could not replicate from Cordova master.
Which Cordova version are you using?

**Comment by fgarcia5** [comment]
Thank you for you feedback @hazems 

These are the details of my environment:
- Android Studio version: 3.3.10
- Gradle version: 4.10.1
- Cordova version: 8.1.2 (cordova-lib@8.1.1)
- Cordova-android version: 7.1.4

I set min and target SDK from Android Studio >> "Project Structure"

When I set target SDK to API 28, the backbutton begins to work correctly after:
- Unfold the keyboard
- Change of application and I return to it
- Blocking and unblocking the cell phone

Any idea what it could be?

Thank you very much

**Comment by tuif** [comment]
the same issue. Look forward to solution.

**Comment by tuif** [comment]
Btw, I'm using cordova-android 8 with default target SDK setting which is 28. Back-button will close the app directly(no 'backbutton' event triggered in JS side) until doing some interaction with APP.
If switching to Cordova-android 7.1.4 with default target SDK setting, everything works fine.

**Comment by rodrigograca31** [comment]
Can't replicate...
```
cordova create hello com.example.hello HelloWorld
cd hello/
cordova platform add android@^7.1.4
cordova run
cordova platform rm android
cordova platform add android@^8.0.0
cordova run
```

my javascript
```javascript
// Application Constructor
initialize: function() {
    document.addEventListener('deviceready', this.onDeviceReady.bind(this), false);
    document.addEventListener("backbutton", this.onBackKeyDown, false);
},

onBackKeyDown: function()
{
    console.log("something");
},
```

Tried 7.1.4, 8.0.0, tried to do the Android Studio thing, tested on Android 8 and 9 and worked 100% as expected. (logged "something")

```
Node version: v8.10.0
Cordova version: 8.1.1
```

@fgarcia5 would you mind running the above commands to create a brand new app, add the javascript and see if the problem persists?

**Comment by fgarcia5** [comment]
Hello @rodrigograca31 , thank you for your feedback.

I have performed the indicated tests and the backbutton works correctly but I have detected that when I add the plugin "cordova-plugin-splashscreen" and set the targetSDK in API 28 (from Android Studio) the backbutton no longer triggers, it kills the app.

Could you do that test and tell me if you find the same bug?

What do you think about?

Regards, Fabricio.

**Comment by MaximeLimone** [comment]
We have the same problem, we have also the cordova-plugin-splashscreen...
   @ionic/cli-utils  : 1.19.2
    ionic (Ionic CLI) : 4.10.3 
    cordova (Cordova CLI) : 8.1.2 
    Cordova Platforms : android 8.0.0 
    Ionic Framework   : ionic1 master

**Comment by hazems** [comment]
I think since it is related to Splash Screen, this ticket can be closed here and a new ticket can be created in https://github.com/apache/cordova-plugin-splashscreen because it does not look like it is a general issue.

**Comment by fgarcia5** [comment]
This issue has been fixed: https://github.com/apache/cordova-plugin-splashscreen/issues/186

**Comment by jgamesl** [comment]
I am also having this issue, weird thing is that the backbutton works again if I resume the app, doesn't make any sense

**Comment by luigi37** [comment]
Same issue here. Sorted with 
https://github.com/prageeth/cordova-plugin-splashscreen

would be great to merge the fix in master...

**Comment by brodybits** [comment]
Let's please keep the discussions in apache/cordova-plugin-splashscreen#186 (issue) and the proposed fix in apache/cordova-plugin-splashscreen#225.

In case of an issue with cordova-android itself, please raise a new issue.

We do not have the time or mental bandwidth to support closed issues.

**Comment by luigi37** [comment]
Uops. Sorry. Had open different pages in browser and wrote comment in the wrong one.

**Comment by imbxl** [comment]
Same issue here. But i'm not using "cordova-plugin-splashscreen", my plugins are:

cordova-plugin-browsersync | npm | ^1.1.0
cordova-plugin-geolocation | npm | ^4.0.2 | 4.0.2
cordova-plugin-inappbrowser | npm | * | 3.2.0
cordova-plugin-network-information | npm | * | 2.0.2
cordova-plugin-whitelist | npm | * | 1.3.4

And my config are:
cli-9.0.0 
Android 8.0.0
